crates/burn-tensor/src/tensor/ops/modules/base.rs

```diff
@@ -114,6 +114,28 @@ pub struct UnfoldOptions {
     pub dilation: [usize; 2],
 }
 
+/// Algorithm used for upsampling.
+pub enum InterpolateMode {
+    /// Nearest-neighbor interpolation.
+    /// <https://en.wikipedia.org/wiki/Nearest-neighbor_interpolation>
+    Nearest,
+
+    /// Bilinear interpolation.
+    /// <https://en.wikipedia.org/wiki/Bilinear_interpolation>
+    Bilinear,
+
+    /// Bicubic interpolation.
+    /// <https://en.wikipedia.org/wiki/Bicubic_interpolation>
+    Bicubic,
+}
+
+/// Interpolation options.
+#[derive(new)]
+pub struct InterpolateOptions {
```

Comment:
I think this is great for future proofing. later on we can add other options like align corners here.